src/backend/metal/src/command.rs

```diff
@@ -129,6 +129,7 @@ pub struct CommandPool {
     shared: Arc<Shared>,
     allocated: Vec<CommandBufferInnerPtr>,
     pool_shared: PoolSharedPtr,
+    layered_rendering: bool,
```

Comment:
The approach of just adding the flags here isn't scalable. Let's make a new type, a bitfield perhaps, something like `EncodingCapabilities`.

Reply:
How about moving `PrivateCapabilities` into `Shared`?